### PR TITLE
Add opencog test runner that works with ctest

### PR DESCRIPTION
**Pull request by AmeBel**
This makes it easy to write scheme tests without having to write c++ codes. See included unit tests for examples.

Depends on https://github.com/opencog/cogutil/pull/131

### CODE REVIEW

File: opencog/scm/opencog/base/utilities.scm [hunk 1]
```diff
@@ -1277,6 +1278,49 @@
...
+    (test-begin \"Name of the test suite\")
+    (test-assert ...)
+    ...
+    ...
+    (test-end \"Name of the test suite\")
+    ; Any code written after 'test-end' is called will not be run as
+    ; 'test-end calls 'exit'.
+
+  For details on the apis for writing srfi-64 test-suites see
+  https://srfi.schemers.org/srfi-64/srfi-64.html
```
Comment (noskill) [inline opencog/scm/opencog/base/utilities.scm]: It looks like the site is down

Comment (linas) [reply opencog/scm/opencog/base/utilities.scm]: It works for me, right now.

**Comment by AmeBel** [comment]
Added the (opencog test-runner) module.

File: opencog/scm/opencog/base/utilities.scm (hunk 1 above)
Comment (noskill) [reply opencog/scm/opencog/base/utilities.scm]: Works for me too

**Pull request merged**